.github/workflows/ci.yaml

```diff
@@ -404,30 +403,29 @@ jobs:
         brew uninstall --ignore-dependencies libffi 2>&1 || true
 
     - name: build/test wheels
-      id: build
+      uses: pypa/cibuildwheel@v2.18.1
       env:
         CIBW_BUILD: ${{ matrix.spec }}
         CIBW_PRERELEASE_PYTHONS: 'True'
         CIBW_TEST_REQUIRES: pytest setuptools
         CIBW_TEST_COMMAND: pip install pip --upgrade; cd {project}; PYTHONUNBUFFERED=1 pytest
         MACOSX_DEPLOYMENT_TARGET: ${{ matrix.deployment_target || '10.9' }}
         SDKROOT: ${{ matrix.sdkroot || 'macosx' }}
-      run: |
-        set -eux
-        
-        mkdir cffi
-        
-        tar zxf ${{ steps.fetch_sdist.outputs.download-path }}/cffi*.tar.gz/cffi*.tar.gz --strip-components=1 -C cffi
```

Comment:
*(with the requested GHA artifact name being non-empty, the new path is actually `${{ steps.fetch_sdist.outputs.download-path }}/cffi*.tar.gz`)*

Reply:
Yeah- the artifact upload/download pathing is painful. It's even worse on Windows, since the GHA context/event properties are *usually* (but not always) Windows paths with drive letters and backslashes, and there aren't enough expression conversion operators to (reliably) convert those. Using bash on Windows for "glue" instead of cmd/PS helps some, but there were still some things that just plain didn't work properly, which is why I had to fall back to not using the artifact fetch paths in all cases.

Reply:
Pretty sure bash can handle those. I had to work around this problem in some action already.